src/hardware/BLEMIDI_ESP32_NimBLE.h

```diff
@@ -128,17 +128,17 @@ bool BLEMIDI_ESP32_NimBLE::begin(const char* deviceName, BLEMIDI_Transport<class
                                                      );
 
     _characteristic->setCallbacks(new MyCharacteristicCallbacks(this));
+
+    auto _security = new NimBLESecurity();
+    _security->setAuthenticationMode(ESP_LE_AUTH_BOND);
+
     // Start the service
     service->start();
-    
-    auto advertisementData = BLEAdvertisementData();
-    advertisementData.setFlags(0x04);
-    advertisementData.setCompleteServices(BLEUUID(SERVICE_UUID));
-    advertisementData.setName(deviceName);
```

Comment:
no longer sets the deviceName

Reply:
Not true, `Device::init` sets the device name.  Even better, this should be `NimBLEDevice::init`